Sage/carbon
# fix(flat-table): move footer to be included in the component height to prevent it hiding table rows

fix #4273

### Proposed behaviour



Removes two unnecessary wrappers `StyledFlatTableRoot` and `StyledFlatTableBox` from `FlatTable`.
Moves `footer` node to be a direct sibling of the table so it is included as part of the component
height. Adds `Pager` to `custom sidebar` `Drawer` story

### Current behaviour


Footer is outside element that has adjustable height, there are unnecessary wrapper elements in the html structure, minHeight prop is not documented, no example of row based min-height

### Checklist




#### QA


### Additional context

![image](https://user-images.githubusercontent.com/44157880/136049825-fb782bb3-e319-48ec-8d02-a00433c39a8c.png)

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


see linked issue for codesandbox and add `minHeight` to `FlatTable`
`docs/flat-table--with-row-based-min-height` added
Worth checking all stories using flat-table for any regressions (including Drawer `custom sidebar`)
